### PR TITLE
Add presentation about method chaining

### PR DESCRIPTION
**Pull request by myz-dev**
Meine Präsentation und einige Aufgaben für den Meetup im September 2024

### CODE REVIEW

File: 2024-09-26/folding-laundry/src/g_tips_and_tricks.rs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            continue;
+        }
+
+        filtered_users.push(u.clone());
+        if filtered_users.len() >= 2 {
+            break;
+        }
+    }
+    filtered_users
+}
```
Comment (AaronErhardt) [inline 2024-09-26/folding-laundry/src/g_tips_and_tricks.rs]: The usecase is a pretty good example for when iterators are great actually:

```rust
    let names = ["Aaron", "Tobi", "Michael", "Albert", "Leon", "Anthony"];

    let filtered: Vec<_> = names
        .iter()
        .inspect(|name| println!("processed: {name}")) // only here to prove that iterators are lazy
        .filter(|name| name.starts_with('A'))
        .take(2)
        .collect();

    println!("{filtered:?}");
```

File: 2024-09-26/folding-laundry/src/g_tips_and_tricks.rs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            continue;
+        }
+        filtered_users.push(u.clone());
+        if filtered_users.len() >= 2 {
+            break;
+        }
+    }
+
+    filtered_users
+}
```
Comment (AaronErhardt) [inline 2024-09-26/folding-laundry/src/g_tips_and_tricks.rs]: Also possible to implement as an iterator with much fewer lines:

```rust
    let names = ["Aaron", "Tobi", "Michael", "Albert", "Leon", "Anthony"];

    let filtered: Vec<_> = names
        .windows(2)
        .inspect(|pair| println!("processed: {pair:?}"))
        .filter(|pair| pair[1].len() >= 6)
        .map(|pair| pair[0])
        .take(2)
        .collect();

    println!("{filtered:?}");
```

P. S. seems like you forgot the "region:" comment in this region.

**Review by AaronErhardt — CHANGES_REQUESTED** [review]
That's an awesome tutorial, sad that I won't be present today.

Sorry for being (potentially) a bit annoying about the examples where you shouldn't use iterators. If you want we can merge this anyway.

**Comment by myz-dev** [comment]
Thank you already for taking the time looking at it!
Really great suggestions, I was so focused on finding a counter example that I did not come up with the `windows` solution...
In my lunch break I will make the changes suggested and will try to think of another example where you would like to use a for loop :D

**Comment by myz-dev** [comment]
I did not have enough time to fix the wrong example where we would not want to use iterators, but I managed to include the great examples you have provided. 
It was all a very rushed so I hope I did not make too many mistakes. I will need to fix the example later.

**Comment by AaronErhardt** [comment]
Should we merge this now, so that it is easier to clone for the others or do you want to work on the examples first?

**Comment by myz-dev** [comment]
Let's merge it please, I will need to provide a fix later this week :-(

**Comment by AaronErhardt** [comment]
No pressure, I think everyone is happy you put in the time to work on this :)

**Pull request merged**